src/lib/settings/GatewayDriver.js

```diff
@@ -179,7 +179,7 @@ class GatewayDriver {
 		validateFunction = validateFunction.bind(this);
 		if (!this.client.methods.util.isObject(schema)) throw 'Schema must be a valid object or left undefined for an empty object.';
 
-		options.provider = this._checkProvider(options.provider || this.client.options.provider.engine || 'json');
+		options.provider = this._checkProvider(options.provider || this.client.options.gateways.defaultProvider.engine || 'json');
```

Comment:
defaultProvider is a string, and there is no need to double default it. (... || 'json')